homeassistant/components/http/static.py

```diff
@@ -60,24 +59,13 @@ async def _handle(self, request: Request) -> StreamResponse:
                 # perm error or other kind!
                 request.app.logger.exception("Unexpected exception")
                 raise HTTPNotFound from error
+            PATH_CACHE[key] = filepath
 
-            content_type: str | None = None
-            if filepath is not None:
-                content_type = (mimetypes.guess_type(rel_url))[
-                    0
-                ] or "application/octet-stream"
-            PATH_CACHE[key] = (filepath, content_type)
-        else:
-            filepath, content_type = filepath_content_type
-
-        if filepath and content_type:
+        if filepath:
             return FileResponse(
                 filepath,
                 chunk_size=self._chunk_size,
-                headers={
-                    hdrs.CACHE_CONTROL: CACHE_HEADER,
-                    hdrs.CONTENT_TYPE: content_type,
```

Comment:
If `hdrs.CONTENT_TYPE` isn't set it looks like the mime type will always be calculated in https://github.com/aio-libs/aiohttp/pull/8485/files#diff-7eeb7201d1ad679d4091189dc43b7a7b843c1b2fc0f0914301addcb25179e6b1R212 if the file is going to be served

Reply:
Correct, that was one of my points above.  For static files, that should be a minimum though (essentially once per client for most files).